WooCommerce/WooCommerceTests/ViewRelated/Dashboard/DashboardViewModelTests.swift

```diff
@@ -403,6 +405,185 @@ final class DashboardViewModelTests: XCTestCase {
+        }
+
+        XCTAssertEqual(viewModel.dashboardCards, expectedCards)
+    }
+
+    func test_generated_default_cards_are_as_expected_with_m2_feature_flag_disabled() {
+        // Given
+        let featureFlagService = MockFeatureFlagService(isDynamicDashboardM2Enabled: false)
+
+        let viewModel = DashboardViewModel(siteID: sampleSiteID, stores: stores, featureFlags: featureFlagService)
+        mockLoadDashboardCards()
+
+        let expectedCards = [DashboardCard(type: .onboarding, availability: .show, enabled: true),
+                             DashboardCard(type: .performance, availability: .unavailable, enabled: false),
+                             DashboardCard(type: .topPerformers, availability: .unavailable, enabled: false),
+                             DashboardCard(type: .blaze, availability: .hide, enabled: false)]
+
+        // When
+        viewModel.refreshDashboardCards()
+
+        // Then
+        waitUntil {
+            viewModel.dashboardCards.isNotEmpty
+        }
+
+        XCTAssertEqual(viewModel.dashboardCards, expectedCards)
+    }
+
+    func test_dashboard_cards_contain_enabled_analytics_cards_when_there_is_order() {
+        // Given
+        let featureFlagService = MockFeatureFlagService(isDynamicDashboardM2Enabled: true)
+        let storage = MockStorageManager()
+        let insertOrder = Order.fake().copy(siteID: sampleSiteID)
+        storage.insertSampleOrder(readOnlyOrder: insertOrder)
+        let viewModel = DashboardViewModel(siteID: sampleSiteID, stores: stores, storageManager: storage, featureFlags: featureFlagService)
+        mockLoadDashboardCards()
+
+        // The expected cards need to be availability: .show and enabled: true
+        let expectedPerformanceCard = DashboardCard(type: .performance, availability: .show, enabled: true)
+        let expectedTopPerformersCard = DashboardCard(type: .topPerformers, availability: .show, enabled: true)
+
+        // When
+        viewModel.refreshDashboardCards()
+
+        // Then
+        waitUntil {
+            viewModel.dashboardCards.isNotEmpty
+        }
+
+        XCTAssertTrue(viewModel.dashboardCards.contains(expectedPerformanceCard))
+        XCTAssertTrue(viewModel.dashboardCards.contains(expectedTopPerformersCard))
+    }
+
+    func test_dashboard_cards_has_disabled_onboarding_card_if_all_tasks_are_completed() {
+        // Given
+        let featureFlagService = MockFeatureFlagService(isDynamicDashboardM2Enabled: false)
+        userDefaults[.completedAllStoreOnboardingTasks] = true
+
+        let viewModel = DashboardViewModel(siteID: sampleSiteID, stores: stores, featureFlags: featureFlagService, userDefaults: userDefaults)
+
+        mockLoadDashboardCards()
+
+        // When
+        viewModel.refreshDashboardCards()
+
+        // Then
+        waitUntil {
+            viewModel.dashboardCards.isNotEmpty
+        }
+
+        XCTAssertFalse(viewModel.dashboardCards.first(where: {$0.type == .onboarding })!.enabled)
```

Comment:
Nit: we can avoid forced unwrapping with `try XCTUnwrap`.

Reply:
Thanks for the reminder! https://github.com/woocommerce/woocommerce-ios/pull/12842/commits/ae190da6d089f9ef510916cb4fe3e097bdfbe1e7